src/icons/fileIcons.ts

```diff
@@ -2397,13 +2397,17 @@ export const fileIcons: FileIcons = {
       name: 'esbuild',
       fileNames: [
         'esbuild.js',
-        'esbuild.ts',
         'esbuild.cjs',
         'esbuild.mjs',
+        'esbuild.ts',
+        'esbuild.cts',
+        'esbuild.mts',
         'esbuild.config.js',
-        'esbuild.config.ts',
         'esbuild.config.cjs',
         'esbuild.config.mjs',
+        'esbuild.config.ts',
+        'esbuild.config.cts',
+        'esbuild.config.mts',
       ],
```

Comment:
Use the `patterns` functionality instead.

Instead of providing all the possible combinations of filenames, we could use `FileNamePattern.Ecmascript` which automatically adds js,cjs,mjs,ts,cts,mts extensions.

In summary, you would need to replace the `fileName` option with the following:

```js
    {
      name: 'esbuild',
      patterns: {
        esbuild: FileNamePattern.Ecmascript,
        'esbuild.config': FileNamePattern.Ecmascript,
      },
    }
```

Reply:
Done!